pio/rs422_tx/rs422_tx.c

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "pico/stdlib.h"
+#include "hardware/pio.h"
+#include "rs422_tx.pio.h"
+
+// Defines the GPIO pins for PIO communication.
+// PIO_TXP_PIN is the positive pin.
+// PIO_TXM_PIN is the negative pin and must be the next pin in sequence.
+#define PIO_TXP_PIN 0
+#define PIO_TXM_PIN 1
+
+// Check the pin is compatible with the platform
+#if (PIO_TXM_PIN) != (PIO_TXP_PIN + 1)
+#error "PIO_TXM_PIN must be the next pin after PIO_TXP_PIN."
+#endif
+#if (PIO_TXM_PIN) >= NUM_BANK0_GPIOS
+#error "Attempting to use a pin>=32 on a platform that does not support it"
+#endif
```

Comment:
Note that `NUM_BANK0_GPIOS` is 30 on RP2040 and RP2350A, and 48 on RP2350B :wink: